src/core/options-watcher.cpp

```diff
@@ -31,18 +30,19 @@ options_watcher::~options_watcher()
 
 void options_watcher::register_option( rs2_option id, std::shared_ptr< option > option )
 {
+    registered_option opt = { option, 0.0f };
+    try
+    {
+        opt.last_known_value = option->query();
```

Comment:
I don't like that we do a query during device initialization...
It's not a problem with the code, but can't we maybe simply recognize that the option doesn't have a previous value and initialize it on the first pass in the thread? That would use a bulk-update (with DDS).

Reply:
We also need to invalidate all the options when we start, right? I.e., we need the query when we start(), not when we register()...

Reply:
Moved first query to `start()`.